flume/src/test/java/org/apache/apex/malhar/flume/sink/FlumeSinkTest.java

```diff
@@ -119,10 +133,10 @@ public void connected()
         array[offset + 8] = 8;
         Server.writeLong(array, offset + Server.Request.TIME_OFFSET, System.currentTimeMillis());
         write(array, offset, Server.Request.FIXED_SIZE);
+        // logger.info("Connect complete");
       }
 
     };
-
     DefaultEventLoop eventloop = new DefaultEventLoop("Eventloop-TestClient");
     eventloop.start();
```

Comment:
Does it make a difference if you move `sink.start()` after `eventloop.start()` ?

Reply:
@tweise  - Moving the event loop did help ( although I have attempted this only once ) . Will have to re-run the test a few times more to confirm this.

Some more information is present in the comment below. Also I have seen some very strange behaviors in some of the builds. Jenkins build failed on the EventLoop as well once when there was a launch of the travis Flume test ( Jenkins does not seem to be configured for running the Flume tests and only Travis does the Flume testing ) but what is more confusing is that the test failed in another component ( I guess it was the JDBCPollTest )

Reply:
Yes, there are other flaky tests that fail the CI sometimes, but not nearly as often as this issue. See also comment WRT logging in `advertise` - your thread dump was valuable to get closer to the root cause.